cpp/include/cudf/binaryop.hpp

```diff
@@ -28,31 +28,35 @@ namespace experimental {
  * @brief Types of binary operations that can be performed on data.
  */
 enum class binary_operator {
-  ADD,             ///< operator +
-  SUB,             ///< operator -
-  MUL,             ///< operator *
-  DIV,             ///< operator / using common type of lhs and rhs
-  TRUE_DIV,        ///< operator / after promoting type to floating point
-  FLOOR_DIV,       ///< operator / after promoting to 64 bit floating point and then
-                   ///< flooring the result
-  MOD,             ///< operator %
-  PYMOD,           ///< operator % but following python's sign rules for negatives
-  POW,             ///< lhs ^ rhs
-  EQUAL,           ///< operator ==
-  NOT_EQUAL,       ///< operator !=
-  LESS,            ///< operator <
-  GREATER,         ///< operator >
-  LESS_EQUAL,      ///< operator <=
-  GREATER_EQUAL,   ///< operator >=
-  BITWISE_AND,     ///< operator &
-  BITWISE_OR,      ///< operator |
-  BITWISE_XOR,     ///< operator ^
-  LOGICAL_AND,     ///< operator &&
-  LOGICAL_OR,      ///< operator ||
-  COALESCE,        ///< operator x,y  x is null ? y : x
-  GENERIC_BINARY,  ///< generic binary operator to be generated with input
-                   ///< ptx code
-  INVALID_BINARY   ///< invalid operation
+  ADD,                    ///< operator +
+  SUB,                    ///< operator -
+  MUL,                    ///< operator *
+  DIV,                    ///< operator / using common type of lhs and rhs
+  TRUE_DIV,               ///< operator / after promoting type to floating point
+  FLOOR_DIV,              ///< operator / after promoting to 64 bit floating point and then
+                          ///< flooring the result
+  MOD,                    ///< operator %
+  PYMOD,                  ///< operator % but following python's sign rules for negatives
+  POW,                    ///< lhs ^ rhs
+  EQUAL,                  ///< operator ==
+  NOT_EQUAL,              ///< operator !=
+  LESS,                   ///< operator <
+  GREATER,                ///< operator >
+  LESS_EQUAL,             ///< operator <=
+  GREATER_EQUAL,          ///< operator >=
+  BITWISE_AND,            ///< operator &
+  BITWISE_OR,             ///< operator |
+  BITWISE_XOR,            ///< operator ^
+  LOGICAL_AND,            ///< operator &&
+  LOGICAL_OR,             ///< operator ||
+  COALESCE,               ///< operator x,y  x is null ? y : x
+  GENERIC_BINARY,         ///< generic binary operator to be generated with input
+                          ///< ptx code
+  SHIFT_LEFT,             ///< operator <<
+  SHIFT_RIGHT,            ///< operator >>
+  SHIFT_RIGHT_UNSIGNED,   ///< operator >>> (from Java)
```

Comment:
This needs standalone documentation about what this is.

Reply:
Couldn't this be implemented by first casting to unsigned and then doing a right shift?

Reply:
Sure.  Where did you have in mind? Inline with the enum value?

Reply:
Yeah, just something like "Cast to unsigned then right shift" or even better, just call it "Logical right shift"

Reply:
> Couldn't this be implemented by first casting the column to unsigned and then doing a right shift?

I am ambivalent. We can say this about all JITted binops which are just a couple of casts away from not requiring JIT. Like, we could do a `int32 x + float y = int64 z` with `(int64)((double)x + (double)y)` but we want to save the casts. OTOH, this case is pretty specialized and we **can** remove an entire operation by this cast.